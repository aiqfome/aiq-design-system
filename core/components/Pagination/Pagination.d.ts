import React from 'react';
export interface Props {
    count?: number;
    color?: string;
    disabled?: boolean;
    variant?: 'default' | 'noCount';
    size?: 'default' | 'small' | 'large';
    defaultPage?: number;
    page?: number;
    nextPage?: any;
    prevPage?: any;
    onChange?: (page: number) => void;
}
export declare const Pagination: React.FC<Props>;
//# sourceMappingURL=Pagination.d.ts.map