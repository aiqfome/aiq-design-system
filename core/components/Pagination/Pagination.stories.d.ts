import React from 'react';
declare const _default: {
    component: React.FC<import("./Pagination").Props>;
    title: string;
};
export default _default;
export declare const Basic: React.FC;
export declare const WithZeroPages: React.FC;
export declare const WithOnePages: React.FC;
export declare const Disabled: React.FC;
export declare const Sizes: React.FC;
export declare const Variants: React.FC;
//# sourceMappingURL=Pagination.stories.d.ts.map