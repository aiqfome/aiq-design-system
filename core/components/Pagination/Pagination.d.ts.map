{"version":3,"file":"Pagination.d.ts","sourceRoot":"","sources":["../../../src/components/Pagination/Pagination.tsx"],"names":[],"mappings":"AAAA,OAAO,KAA8B,MAAM,OAAO,CAAA;AAUlD,MAAM,WAAW,KAAK;IACpB,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,QAAQ,CAAC,EAAE,OAAO,CAAA;IAClB,OAAO,CAAC,EAAE,SAAS,GAAG,SAAS,CAAA;IAC/B,IAAI,CAAC,EAAE,SAAS,GAAG,OAAO,GAAG,OAAO,CAAA;IACpC,WAAW,CAAC,EAAE,MAAM,CAAA;IACpB,IAAI,CAAC,EAAE,MAAM,CAAA;IACb,QAAQ,CAAC,EAAE,GAAG,CAAA;IACd,QAAQ,CAAC,EAAE,GAAG,CAAA;IACd,QAAQ,CAAC,EAAE,CAAC,IAAI,EAAE,MAAM,KAAK,IAAI,CAAA;CAClC;AAoBD,eAAO,MAAM,UAAU,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAqNtC,CAAA"}