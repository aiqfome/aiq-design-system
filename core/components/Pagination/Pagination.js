"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pagination = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const md_1 = require("react-icons/md");
const styled_components_1 = __importStar(require("styled-components"));
const Flex_1 = require("../Flex");
const Text_1 = require("../Text");
const PaginationItem_1 = require("./PaginationItem");
const PaginationStyled = styled_components_1.default(Flex_1.Flex) `
  flex-wrap: wrap;

  ${({ disabled }) => disabled &&
    styled_components_1.css `
      opacity: 0.5;
      &:hover {
        pointer-events: none;
      }

      & > Button,
      & span {
        cursor: not-allowed !important;
      }
    `}
`;
exports.Pagination = (_a) => {
    var { size, variant, count = 0, defaultPage = 1, nextPage, prevPage, disabled = false, onChange = () => {
        // do nothing.
    } } = _a, props = __rest(_a, ["size", "variant", "count", "defaultPage", "nextPage", "prevPage", "disabled", "onChange"]);
    const [pages, setPages] = react_1.useState([]);
    const [pagesToShow, setPagesToShow] = react_1.useState([]);
    const [currentPage, setCurrentPage] = react_1.useState(1);
    const range = (start, end) => {
        const length = end - start + 1;
        return Array.from({ length }, (_, i) => start + i);
    };
    react_1.useEffect(() => {
        const newPages = range(1, count);
        setPages(newPages);
    }, [count]);
    react_1.useEffect(() => {
        setCurrentPage(defaultPage);
    }, [defaultPage]);
    react_1.useEffect(() => {
        let startPage = currentPage - 2;
        let endPage = currentPage + 2;
        if (variant === 'default') {
            if (startPage <= 1) {
                endPage -= startPage - 1;
                startPage = 1;
            }
            if (endPage > count - 2) {
                startPage = count - 5 || 1;
                endPage = count - 1;
            }
            if (startPage <= 1) {
                startPage = 1;
            }
            setPagesToShow(pages.slice(startPage, endPage));
        }
        else {
            endPage = currentPage + 1;
            startPage = currentPage - 1;
            if (startPage <= 1) {
                startPage = 2;
                endPage = startPage + 1;
            }
            setPagesToShow(range(startPage, endPage));
        }
    }, [pages, count, variant, currentPage]);
    function handleClickPage(page) {
        if (page > 0 && (variant === 'noCount' || (count && page <= count))) {
            setCurrentPage(page);
            onChange(page);
        }
    }
    return (react_1.default.createElement(PaginationStyled, Object.assign({ alignItems: 'center', flexDirection: 'row', disabled: disabled, justifyContent: 'center' }, props),
        variant === 'default' && count > 0 && (react_1.default.createElement(react_1.default.Fragment, null,
            react_1.default.createElement(PaginationItem_1.PaginationItem, { "data-testid": 'pagination-item-return', size: size, cursor: currentPage === 1 ? 'not-allowed' : '', onClick: disabled ? undefined : () => handleClickPage(currentPage - 1) },
                react_1.default.createElement(md_1.MdChevronLeft, null)),
            react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, "data-testid": 'pagination-item-1', active: currentPage === 1, onClick: disabled ? undefined : () => handleClickPage(1) },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, "1")),
            currentPage - 2 > 0 && count > 6 && (react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, cursor: 'auto' },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, "..."))),
            pagesToShow.map(page => (react_1.default.createElement(PaginationItem_1.PaginationItem, { "data-testid": `pagination-item-${page}`, size: size, key: page.toString(), active: page === currentPage, onClick: disabled ? undefined : () => handleClickPage(page) },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, page)))),
            currentPage + 3 < pages[pages.length - 1] && count > 6 && (react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, cursor: 'auto' },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, "..."))),
            pages.length > 1 && (react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, "data-testid": `pagination-item-${pages[pages.length - 1]}`, active: currentPage === pages[pages.length - 1], onClick: disabled
                    ? undefined
                    : () => handleClickPage(pages[pages.length - 1]) },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, pages[pages.length - 1]))),
            react_1.default.createElement(PaginationItem_1.PaginationItem, { "data-testid": 'pagination-item-next', size: size, onClick: disabled ? undefined : () => handleClickPage(currentPage + 1), cursor: currentPage === pages[pages.length - 1] ? 'not-allowed' : '' },
                react_1.default.createElement(md_1.MdChevronRight, null)))),
        variant === 'noCount' && (react_1.default.createElement(react_1.default.Fragment, null,
            react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, "data-testid": 'pagination-item-noCunt-return', cursor: currentPage === 1 ? 'not-allowed' : '', onClick: () => handleClickPage(currentPage - 1), disabled: !prevPage },
                react_1.default.createElement(md_1.MdChevronLeft, null)),
            react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, active: currentPage === 1, onClick: disabled ? undefined : () => handleClickPage(1) },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, "1")),
            currentPage - 2 > 0 && (react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, cursor: 'auto' },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, "..."))),
            currentPage > 1 && (react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, active: true },
                react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'default' }, currentPage))),
            react_1.default.createElement(PaginationItem_1.PaginationItem, { size: size, "data-testid": 'pagination-item-noCunt-next', onClick: () => handleClickPage(currentPage + 1), cursor: !nextPage ? 'not-allowed' : '', disabled: !nextPage },
                react_1.default.createElement(md_1.MdChevronRight, null))))));
};
exports.Pagination.defaultProps = {
    variant: 'default'
};
exports.Pagination.propTypes = {
    page: prop_types_1.default.number,
    count: prop_types_1.default.number,
    color: prop_types_1.default.string,
    onChange: prop_types_1.default.func,
    disabled: prop_types_1.default.bool,
    defaultPage: prop_types_1.default.number,
    size: prop_types_1.default.oneOf(['default', 'small', 'large']),
    variant: prop_types_1.default.oneOf(['default', 'noCount']),
    nextPage: prop_types_1.default.any,
    prevPage: prop_types_1.default.any
};
