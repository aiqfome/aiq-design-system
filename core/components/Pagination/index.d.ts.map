{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Pagination/index.tsx"],"names":[],"mappings":"AAAA,cAAc,cAAc,CAAA"}