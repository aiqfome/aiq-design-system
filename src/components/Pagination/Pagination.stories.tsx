import React from 'react'

import { Flex } from '../Flex'
import { Pagination } from './Pagination'

export default {
  component: Pagination,
  title: 'Pagination'
}

export const Basic: React.FC = () => {
  return <Pagination count={10} />
}

export const WithZeroPages: React.FC = () => {
  return <Pagination count={0} />
}

export const WithOnePages: React.FC = () => {
  return <Pagination count={1} />
}

export const Disabled: React.FC = () => {
  return <Pagination disabled count={10} />
}

export const Sizes: React.FC = () => {
  return (
    <>
      <Flex mb='20px' variant='centralized'>
        <Pagination count={10} size='small' />
      </Flex>

      <Flex mb='20px' variant='centralized'>
        <Pagination count={10000} size='default' />
      </Flex>

      <Flex variant='centralized'>
        <Pagination count={10} size='large' />
      </Flex>
    </>
  )
}

export const Variants: React.FC = () => {
  return (
    <>
      <Flex variant='centralized' mb='20px'>
        <Pagination variant='noCount' size='large' />
      </Flex>

      <Flex variant='centralized'>
        <Pagination count={100} size='large' />
      </Flex>
    </>
  )
}
