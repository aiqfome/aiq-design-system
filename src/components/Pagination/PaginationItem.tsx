import React from 'react'

import PropTypes from 'prop-types'

import styled, { css, DefaultTheme } from 'styled-components'

import { Button } from '../Button'

interface Props extends DefaultTheme {
  active?: boolean
  children?: any
  cursor?: string
  className?: string
  disabled?: boolean
  size?: 'default' | 'small' | 'large'
  onClick?: () => void
}

const ItemPageStyled = styled(Button)<Props>`
  &.__pagination-size-large {
    height: 42px;
    min-width: 42px;
  }
  &.__pagination-size-default {
    height: 34px;
    min-width: 34px;
  }
  &.__pagination-size-small {
    height: 28px;
    min-width: 28px;
  }

  padding: 0 8px;
  border: 1px solid #dedede;
  border-radius: 0;
  justify-content: center;
  align-items: center;

  &:first-child {
    border-radius: 3px 0 0 3px;
  }

  &:last-child {
    border-radius: 0 3px 3px 0;
  }

  &:hover {
    outline: none;
    text-decoration: none;
  }

  cursor: ${({ cursor }) => cursor || 'pointer'};

  & span {
    cursor: ${({ cursor }) => cursor || 'pointer'};
  }

  ${({ active, theme }) =>
    active &&
    css`
      background: ${theme.colors.primary};
      border: 1px solid ${theme.colors.primary};

      span {
        color: ${theme.colors.white};
      }
    `}
`

export const PaginationItem: React.FC<Props> = ({
  children,
  size = 'default',
  ...props
}) => {
  return (
    <ItemPageStyled
      className={`__pagination-item __pagination-size-${size}`}
      size={size}
      {...props}
    >
      {children}
    </ItemPageStyled>
  )
}

PaginationItem.propTypes = {
  children: PropTypes.any,
  size: PropTypes.oneOf(['default', 'small', 'large'])
}
