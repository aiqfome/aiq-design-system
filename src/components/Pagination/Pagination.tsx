import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { MdChevronLeft, MdChevronRight } from 'react-icons/md'
import styled, { css } from 'styled-components'

import { Flex } from '../Flex'
import { Text } from '../Text'

import { PaginationItem } from './PaginationItem'

export interface Props {
  count?: number
  color?: string
  disabled?: boolean
  variant?: 'default' | 'noCount'
  size?: 'default' | 'small' | 'large'
  defaultPage?: number
  page?: number
  onChange?: (page: number) => void
}

const PaginationStyled = styled(Flex)<Props>`
  flex-wrap: wrap;

  ${({ disabled }) =>
    disabled &&
    css`
      opacity: 0.5;
      &:hover {
        pointer-events: none;
      }

      & > Button,
      & span {
        cursor: not-allowed !important;
      }
    `}
`

export const Pagination: React.FC<Props> = ({
  size,
  variant,
  count = 0,
  defaultPage = 1,
  disabled = false,
  onChange = () => {
    // do nothing.
  },
  ...props
}) => {
  const [pages, setPages] = useState<number[]>([])
  const [pagesToShow, setPagesToShow] = useState<number[]>([])
  const [currentPage, setCurrentPage] = useState(1)

  const range = (start, end) => {
    const length = end - start + 1
    return Array.from({ length }, (_, i) => start + i)
  }

  useEffect(() => {
    const newPages = range(1, count)
    setPages(newPages)
  }, [count])

  useEffect(() => {
    setCurrentPage(defaultPage)
  }, [defaultPage])

  useEffect(() => {
    let startPage = currentPage - 2
    let endPage = currentPage + 2

    if (variant === 'default') {
      if (startPage <= 1) {
        endPage -= startPage - 1
        startPage = 1
      }

      if (endPage > count - 2) {
        startPage = count - 5 || 1
        endPage = count - 1
      }

      if (startPage <= 1) {
        startPage = 1
      }

      setPagesToShow(pages.slice(startPage, endPage))
    } else {
      endPage = currentPage + 1
      startPage = currentPage - 1

      if (startPage <= 1) {
        startPage = 2
        endPage = startPage + 1
      }

      setPagesToShow(range(startPage, endPage))
    }
  }, [pages, count, variant, currentPage])

  function handleClickPage(page) {
    if (page > 0 && (variant === 'noCount' || (count && page <= count))) {
      setCurrentPage(page)
      onChange(page)
    }
  }

  return (
    <PaginationStyled
      alignItems='center'
      flexDirection='row'
      disabled={disabled}
      justifyContent='center'
      {...props}
    >
      {variant === 'default' && count > 0 && (
        <>
          <PaginationItem
            data-testid='pagination-item-return'
            size={size}
            cursor={currentPage === 1 ? 'not-allowed' : ''}
            onClick={
              disabled ? undefined : () => handleClickPage(currentPage - 1)
            }
          >
            <MdChevronLeft />
          </PaginationItem>

          <PaginationItem
            size={size}
            data-testid='pagination-item-1'
            active={currentPage === 1}
            onClick={disabled ? undefined : () => handleClickPage(1)}
          >
            <Text color='almostBlack' fontSize='default'>
              1
            </Text>
          </PaginationItem>
          {currentPage - 2 > 0 && count > 6 && (
            <PaginationItem size={size} cursor='auto'>
              <Text color='almostBlack' fontSize='default'>
                ...
              </Text>
            </PaginationItem>
          )}

          {pagesToShow.map(page => (
            <PaginationItem
              data-testid={`pagination-item-${page}`}
              size={size}
              key={page.toString()}
              active={page === currentPage}
              onClick={disabled ? undefined : () => handleClickPage(page)}
            >
              <Text color='almostBlack' fontSize='default'>
                {page}
              </Text>
            </PaginationItem>
          ))}
          {currentPage + 3 < pages[pages.length - 1] && count > 6 && (
            <PaginationItem size={size} cursor='auto'>
              <Text color='almostBlack' fontSize='default'>
                ...
              </Text>
            </PaginationItem>
          )}

          {pages.length > 1 && (
            <PaginationItem
              size={size}
              data-testid={`pagination-item-${pages[pages.length - 1]}`}
              active={currentPage === pages[pages.length - 1]}
              onClick={
                disabled
                  ? undefined
                  : () => handleClickPage(pages[pages.length - 1])
              }
            >
              <Text color='almostBlack' fontSize='default'>
                {pages[pages.length - 1]}
              </Text>
            </PaginationItem>
          )}
          <PaginationItem
            data-testid='pagination-item-next'
            size={size}
            onClick={
              disabled ? undefined : () => handleClickPage(currentPage + 1)
            }
            cursor={
              currentPage === pages[pages.length - 1] ? 'not-allowed' : ''
            }
          >
            <MdChevronRight />
          </PaginationItem>
        </>
      )}

      {variant === 'noCount' && (
        <>
          <PaginationItem
            size={size}
            data-testid='pagination-item-noCunt-return'
            cursor={currentPage === 1 ? 'not-allowed' : ''}
            onClick={
              disabled ? undefined : () => handleClickPage(currentPage - 1)
            }
          >
            <MdChevronLeft />
          </PaginationItem>

          <PaginationItem
            size={size}
            active={currentPage === 1}
            onClick={disabled ? undefined : () => handleClickPage(1)}
          >
            <Text color='almostBlack' fontSize='default'>
              1
            </Text>
          </PaginationItem>

          {currentPage - 2 > 0 && (
            <PaginationItem size={size} cursor='auto'>
              <Text color='almostBlack' fontSize='default'>
                ...
              </Text>
            </PaginationItem>
          )}

          {pagesToShow.map(page => (
            <PaginationItem
              size={size}
              key={page.toString()}
              active={page === currentPage}
              onClick={disabled ? undefined : () => handleClickPage(page)}
            >
              <Text color='almostBlack' fontSize='default'>
                {page}
              </Text>
            </PaginationItem>
          ))}

          <PaginationItem
            size={size}
            data-testid='pagination-item-noCunt-next'
            onClick={
              disabled ? undefined : () => handleClickPage(currentPage + 1)
            }
            cursor={
              currentPage === pages[pages.length - 1] ? 'not-allowed' : ''
            }
          >
            <MdChevronRight />
          </PaginationItem>
        </>
      )}
    </PaginationStyled>
  )
}

Pagination.defaultProps = {
  variant: 'default'
}

Pagination.propTypes = {
  page: PropTypes.number,
  count: PropTypes.number,
  color: PropTypes.string,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  defaultPage: PropTypes.number,
  size: PropTypes.oneOf(['default', 'small', 'large']),
  variant: PropTypes.oneOf(['default', 'noCount'])
}
