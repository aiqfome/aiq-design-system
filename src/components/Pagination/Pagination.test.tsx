import React from 'react'

import { fireEvent } from '@testing-library/react'

import { render } from '../utils/test/render'

import { Pagination } from '../Pagination'

describe('Pagination', () => {
  it('should have to render without crashing', () => {
    const component = render(<Pagination />)
    expect(component).toBeTruthy()
  })

  it('should not render when count is zero', () => {
    const { container } = render(<Pagination count={0} />)
    const paginationsItens = container.querySelectorAll('.__pagination-item')

    expect(paginationsItens.length).toBe(0)
  })

  it('should render only a page when count is 1', () => {
    const { container } = render(<Pagination count={1} />)
    const paginationsItens = container.querySelectorAll('.__pagination-item')

    expect(paginationsItens.length).toBe(3)
  })

  it('should call onChange when click in next page or return page ', () => {
    const handleOnChange = jest.fn()

    const { getByTestId } = render(
      <Pagination count={10} onChange={handleOnChange} />
    )

    const itemNext = getByTestId('pagination-item-next')
    const itemReturn = getByTestId('pagination-item-return')
    const itemPage10 = getByTestId('pagination-item-10')

    fireEvent.click(itemNext)
    expect(handleOnChange).toHaveBeenLastCalledWith(2)
    fireEvent.click(itemPage10)
    expect(handleOnChange).toHaveBeenLastCalledWith(10)
    fireEvent.click(itemReturn)
    expect(handleOnChange).toHaveBeenLastCalledWith(9)
    expect(handleOnChange).toHaveBeenCalledTimes(3)
  })

  it('should not call onChange when click in next page or return page and paginations is disabled', () => {
    const handleOnChange = jest.fn()

    const { getByTestId } = render(
      <Pagination count={10} disabled={true} onChange={handleOnChange} />
    )

    const itemNext = getByTestId('pagination-item-next')
    const itemReturn = getByTestId('pagination-item-return')
    const itemPage10 = getByTestId('pagination-item-10')

    fireEvent.click(itemNext)
    fireEvent.click(itemPage10)
    fireEvent.click(itemReturn)

    expect(handleOnChange).toHaveBeenCalledTimes(0)
  })

  it('should have class __pagination-size-default when size is null', () => {
    const { container } = render(<Pagination count={10} />)
    const paginatorItem = container.querySelector('.__pagination-item')

    expect(paginatorItem).toHaveClass('__pagination-size-default')
  })

  it('should have class __pagination-size-small when size is small', () => {
    const { container } = render(<Pagination size={'small'} count={10} />)
    const paginatorItem = container.querySelector('.__pagination-item')

    expect(paginatorItem).toHaveClass('__pagination-size-small')
  })

  it('should have class __pagination-size-large when size is large', () => {
    const { container } = render(<Pagination size='large' count={10} />)
    const paginatorItem = container.querySelector('.__pagination-item')

    expect(paginatorItem).toHaveClass('__pagination-size-large')
  })

  it('should have infinite pages when variant is noCount', () => {
    const handleOnChange = jest.fn()

    const { getByTestId } = render(
      <Pagination variant='noCount' onChange={handleOnChange} />
    )

    const itemNext = getByTestId('pagination-item-noCunt-next')

    let index = 0
    while (index < 100) {
      index++
      fireEvent.click(itemNext)
    }
    expect(handleOnChange).toHaveBeenLastCalledWith(101)
  })

  it('should not have infinite pages when variant is default', () => {
    const handleOnChange = jest.fn()

    const { getByTestId } = render(
      <Pagination count={8} onChange={handleOnChange} />
    )

    const itemNext = getByTestId('pagination-item-next')

    let index = 0
    while (index < 100) {
      index++
      fireEvent.click(itemNext)
    }
    expect(handleOnChange).not.toHaveBeenLastCalledWith(101)
    expect(handleOnChange).toHaveBeenLastCalledWith(8)
  })
})
