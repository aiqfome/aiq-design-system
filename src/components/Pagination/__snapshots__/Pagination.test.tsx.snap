// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxjAm sc-AxhUy fkyrdw"
  disabled={false}
>
  <button
    className="sc-AxhCb jGTOcA sc-AxgMl kAQFoy"
    cursor="not-allowed"
    onClick={[Function]}
  >
    <span
      className="sc-AxirZ cuGRia"
      cursor="pointer"
      fontSize="medium"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </span>
  </button>
  <button
    className="sc-AxhCb jGTOcA sc-AxgMl etdRWP"
    onClick={[Function]}
  >
    <span
      className="sc-AxirZ cuGRia"
      cursor="pointer"
      fontSize="medium"
    >
      <span
        className="sc-AxirZ dNjvtV"
        color="almostBlack"
        fontSize="default"
      >
        1
      </span>
    </span>
  </button>
  <button
    className="sc-AxhCb jGTOcA sc-AxgMl klYcia"
    cursor=""
    onClick={[Function]}
  >
    <span
      className="sc-AxirZ cuGRia"
      cursor="pointer"
      fontSize="medium"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </span>
  </button>
</div>
`;
